src/test/scala/com/microsoft/hyperspace/actions/CreateActionTest.scala

```diff
@@ -110,55 +111,53 @@ class CreateActionTest extends SparkFunSuite with SparkInvolvedSuite {
   }
 
   test("Verify rootPaths for given LogicalRelations") {
-    val path1 = sampleParquetDataLocation + "table1"
-    val path2 = sampleParquetDataLocation + "table2"
-    FileUtils.delete(new Path(path1))
-    FileUtils.delete(new Path(path2))
-
-    import spark.implicits._
-    SampleData.testData
-      .toDF("Date", "RGUID", "Query", "imprs", "clicks")
-      .limit(2)
-      .write
-      .parquet(path1)
-
-    SampleData.testData
-      .toDF("Date", "RGUID", "Query", "imprs", "clicks")
-      .limit(3)
-      .write
-      .parquet(path2)
-
-    val workDir = FileContext.getFileContext.getWorkingDirectory.toString
-
-    Seq(
-      (spark.read.format("parquet").load(path1), Seq(path1), 2),
-      (spark.read.format("parquet").load(path1, path2), Seq(path1, path2), 5),
-      (spark.read.parquet(path1), Seq(path1), 2),
-      (spark.read.parquet(path1, path2), Seq(path1, path2), 5),
-      (spark.read.parquet(path1, path1, path1), Seq(path1, path1, path1), 6),
-      (spark.read.format("parquet").option("path", path1).load(path1), Seq(path1), 2),
-      (spark.read.format("parquet").option("path", path1).load(path2), Seq(path2), 3),
-      (spark.read.option("path", path1).parquet(path1), Seq(path1, path1), 4),
-      (spark.read.option("path", path1).parquet(path2), Seq(path1, path2), 5),
-      (
-        spark.read.format("parquet").option("path", path1).load(path1, path2),
-        Seq(path1, path1, path2),
-        7),
-      (spark.read.option("path", path1).parquet(path1, path2), Seq(path1, path1, path2), 7))
-      .foreach {
-        case (df, expectedRootPaths, expectedCount) =>
-          val relation = CreateActionBaseWrapper.getSourceRelations(df)(0)
-          val rootPaths = relation.rootPaths.map {
-            case path if path.startsWith(workDir) => path.drop(workDir.size + 1) // slash
-            case path => path
-          }
-          assert(rootPaths == expectedRootPaths)
-          assert(df.count == expectedCount)
-          assert(!relation.options.isDefinedAt("path"))
-        case _ => fail("invalid test")
-      }
-
-    FileUtils.delete(new Path(path1))
-    FileUtils.delete(new Path(path2))
+    withTempPath { p =>
+      val path1 = p + "table1"
+      val path2 = p + "table2"
+
+      FileUtils.delete(new Path(path1))
```

Comment:
we can remove these `delete`s here and at the end; that was the reason for using `withTempPath`.